pkg/client/record/event.go

```diff
@@ -42,49 +55,82 @@ type EventRecorder interface {
 // your binary. Subsequent calls will be ignored. The return value can be ignored
 // or used to stop recording, if desired.
 func StartRecording(recorder EventRecorder, source api.EventSource) watch.Interface {
+	// Set up our own personal buffer of events so that we can clear out GetEvents'
+	// broadcast channel as quickly as possible to avoid causing the relatively more
```

Comment:
If you're concerned about this, I think this is the wrong place to fix it, I think you should fix it at the other end of the queue, where Event() stuffs events in.

The way you have it, anyone who calls GetEvents could theoretically block events.

It's also a bit weird to have two 1000 element queues.